src/lib/components/forms/ManageCollaborators.svelte

```diff
@@ -0,0 +1,178 @@
+<!-- @component 
+Manage users on a project: add, update, remove
+
+This component is actually two separate forms that travel together.
+They might get separated later.
+-->
+<script lang="ts">
+  import type {
+    Project,
+    ProjectAccess,
+    ProjectUser,
+    User,
+  } from "$lib/api/types";
+
+  import { enhance } from "$app/forms";
+
+  import { createEventDispatcher } from "svelte";
+  import { _ } from "svelte-i18n";
+
+  import Button from "../common/Button.svelte";
+  import Field from "../inputs/Field.svelte";
+  import Flex from "../common/Flex.svelte";
+  import Text from "../inputs/Text.svelte";
+
+  import { canonicalUrl } from "$lib/api/projects";
+
+  export let project: Project;
+  export let users: ProjectUser[];
+
+  const dispatch = createEventDispatcher();
+
+  $: canonical = canonicalUrl(project);
+  $: action = new URL("?/users", canonical).href;
+  $: invite = new URL("?/invite", canonical).href;
+  $: grouped = users.reduce(
+    (m, { user, access }) => {
+      m[access].push(user);
+      return m;
+    },
+    { admin: [], edit: [], view: [] } satisfies Record<ProjectAccess, User[]>,
+  );
+
+  function sort(users: User[]) {
+    return users.sort((a, b) => a.name.localeCompare(b.name));
+  }
+
+  function onSubmit({ formElement, formData, action, cancel, submitter }) {
+    submitter.disabled = true;
+    return async ({ result, update }) => {
+      submitter.disabled = false;
+      return update(result);
+    };
+  }
+</script>
+
+<Flex direction="column">
+  <form action={invite} method="post" use:enhance>
+    <h2>{$_("collaborators.addCollaborators")}</h2>
+    <p>{@html $_("collaborators.invite")}</p>
+    <Flex align="center">
+      <Field title={$_("common.emailAddress")} sronly>
+        <Text name="email" placeholder={$_("common.emailAddress")} />
+      </Field>
+      <Field>
+        <select name="access" class="access">
+          <option value="view">{$_("collaborators.view")}</option>
+          <option value="edit">{$_("collaborators.edit")}</option>
+          <option value="admin">{$_("collaborators.admin")}</option>
+        </select>
+      </Field>
+      <Button mode="primary" type="submit">
+        + {$_("collaborators.add")}
+      </Button>
+    </Flex>
+  </form>
+
+  {#if users.length}
+    <form {action} method="post" use:enhance>
+      {#each Object.entries(grouped) as [access, group]}
+        {#if group.length}
+          <Flex direction="column" class="users {access}">
+            <h2>{$_(`collaborators.${access}`)}</h2>
+            <p class="help">
+              {$_(`collaborators.help.${access}`)}
+            </p>
+            <table>
+              <tr>
+                <th>{$_("collaborators.name")}</th>
+                <th>{$_("collaborators.access")}</th>
+              </tr>
+              {#each sort(group) as user}
+                <tr>
+                  <td>{user.name}</td>
```

Comment:
Again, we should use `UserAvatar` alongside user names to improve the scannability of collaborator lists:
```suggestion
                  <td><Flex><UserAvatar {user} /> {user.name}</Flex></td>
```